cmd/credential.go

```diff
@@ -50,15 +49,7 @@ type fdoTpmDeviceCredential struct {
 }
 
 func tpmCred() (hash.Hash, hash.Hash, crypto.Signer, func() error, error) {
-	var diKeyFlagSet bool
-	clientFlags.Visit(func(flag *flag.Flag) {
-		if flag == nil {
-			slog.Error("Unexpected nil flag encountered")
-			return
-		}
-		diKeyFlagSet = diKeyFlagSet || flag.Name == "di-key"
-	})
-	if !diKeyFlagSet {
+	if diKey == "" {
```

Comment:
Use Flags().Changed() API here to determine if the user explicitly set the DI key parameter.

Reply:
Let's add a note that we're checking that the flag was explicitly set, because for TPM's it's very important to know what kind of key you're creating. Resources are limited.